src/Repository/Schedule/ScheduleRepositoryManager.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repository\Schedule;
+
+final class ScheduleRepositoryManager
+{
+    private $repository;
+
+    public function __construct(ScheduleRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function countSlotTypes(array $schedules = null): array
+    {
+        if (null === $schedules) {
+            $schedules = $this->repository->findScheduleAndSlots();
+        }
+
+        $slotsTypes = [];
+
+        foreach ($schedules as $schedule) {
+            $slotId = $schedule->getId();
+
+            if (!isset($slotsTypes[$slotId])) {
+                $slotsTypes[$slotId] = [];
+            }
+
+            foreach ($schedule->getSpaces() as $space) {
+                foreach ($space->getSlots() as $slot) {
```

Comment:
argh too many foreach in foreach, can you decouple in private functions please? :)